src/axom/primal/operators/intersection_volume.hpp

```diff
@@ -147,13 +167,18 @@ AXOM_HOST_DEVICE T intersection_volume(const Tetrahedron<T, 3>& tet,
  * \param [in] checkSign If true (default is false), checks if the
```

Comment:
The wording is still weird. Here's a suggestion that's a little simpler.

If true, takes each shape with a negative signed volume and swaps the order of some vertices in that shape to try to obtain a nonnegative signed volume. Defaults to false.

I would add the rest to a warning tag.

Reply:
warning If tryFixOrientation flag is false and some of the shapes have a negative signed volume, the computed volume of intersection may be zero and/or unexpected.

Reply:
I'm still wondering if this flag should default to true.

Reply:
Is there a technical term for valid formats? It seems like only left-handed and right-handed orientations (windings?) are valid. And the checkSign really just converts from a left handed coordinate system to a right handed coordinate system. I wonder if documenting in terms like that would actually simplify the documentation.

Reply:
If true, converts each shape using a left-handed coordinate system to a right-handed coordinate system. Defaults to false.

Reply:
All this duplication makes me wonder if we should just require the caller to pass in shapes with a right-handed coordinate system, particularly now that there is a convenience function to fix the orientation. Best left for another time, though.

Reply:
Sorry for the brain dump :)

Reply:
> Sorry for the brain dump :)

👍 👍   Thanks for the thoughts and wording suggestions! 

> Is there a technical term for valid formats?

Not that I am aware of. I was thinking of "degenerate" and "non-degenerate", but the term is for a specific case of polyhedra.

>  It seems like only left-handed and right-handed orientations (windings?) are valid... I wonder if documenting in terms like that would actually simplify the documentation.

Yes, that is what I think of when I say "valid", and modifying the documentation to fit that makes sense now that I think about it.

Reply:
Hm, thinking about things over again, I think _ideally_ the `tryFixOrientation` flag checks what coordinate system is being used, and swaps vertices around if it's a left-handed. Currently, that is not what is being done, and it's a sign check on the volume. So, the documentation for `tryFixOrientation` should still be in terms of volume and not right-handedness/left-handedness.

A coordinate system check would go beyond just the sign of the volume, and would be able to verify if the ordering is valid, throwing an error or warning for cases like [this](https://github.com/LLNL/axom/pull/1212#discussion_r1386909807) or that:
```
   *          4--------5          +z
   *         /|       /|               +y
   *        / |      / |           ^  >
   *       7--------6  |           | /
   *       |  0-----|--3           |/
   *       | /      | /            -----> +x
   *       |/       |/
   *       1--------2
   *
```

Reply:
@adayton1 wrote:
> Is there a technical term for valid formats? It seems like only left-handed and right-handed orientations (windings?) are valid. And the checkSign really just converts from a left handed coordinate system to a right handed coordinate system (if I understand correctly). I wonder if documenting in terms like that would actually simplify the documentation.

There are three possible orientations for a tetrahedron, two of which are non-degenerate -- the signed volume could be positive, negative or zero. We consider the latter to be degenerate since its four vertices all lie in the same plane. There are some use-cases where negative orientations/volumes are useful, e.g. you can compute the volume of a polyhedron by creating pyramids out of its faces w.r.t. any arbitrary point and summing their volumes.

On the other hand, some mesh generators are not careful about ensuring that all tetrahedra in a collection are consistently oriented, so it can be useful to fix the orientation in a preprocessing step when downstream algorithms assume consistent orientations

I'm not sure if "right-handed" vs. "left-handed" is the best terminology for shapes other than tetrahedra though. E.g. there are a few popular conventions for labeling/ordering the vertices of a hexahedron